### PR TITLE
test_ui_tools: Add failing test for nested blockquote.

### PR DESCRIPTION
**Pull request by sumanthvrao**
Address the failing bug with a test which is marked to fail(#402)

### CODE REVIEW

**Comment by neiljp** [comment]
@sumanthvrao Does this actually contain the source which triggers #402 yet? It looks like what we may have discussed privately, but not the extra case there?

**Comment by sumanthvrao** [comment]
@neiljp Not really. But, fundamentally both this and the one you mentioned in the #test-here boil down to a common problem (bug) in the code wherein a `child` in the `child_list` does not get padded properly, if it itself is a blockquote.

File: tests/ui/test_ui_tools.py [hunk 1]
```diff
@@ -1302,13 +1302,21 @@ def test_keypress_edit_message(self, mocker, message_fixture,
...
+                        </blockquote>
+            """, "░ ░ A\n░ ░ B", marks=pytest.mark.xfail(reason="nested bq"))
     ], ids=[
         "quoted level 1",
         "quoted level 2",
         "quoted level 3",
         "multi-line quoting",
-        "quoting with links"
+        "quoting with links",
+        "nested blockquote"
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: Is this not more like "quoted multiline level 2"?  (though we could likely skip the 'quoted' in the ids)

File: tests/ui/test_ui_tools.py [hunk 2]
```diff
@@ -1302,13 +1302,21 @@ def test_keypress_edit_message(self, mocker, message_fixture,
...
                 <p><a href='https://chat.zulip.org/'</a>czo</p>
-            </blockquote>""", "░ [czo](https://chat.zulip.org/)\n")
+            </blockquote>""", "░ [czo](https://chat.zulip.org/)\n"),
+        pytest.param("""<blockquote>
+                            <blockquote>
+                                <p>A<br>
+                                B</p>
+                            </blockquote>
+                        </blockquote>
+            """, "░ ░ A\n░ ░ B", marks=pytest.mark.xfail(reason="nested bq"))
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: I would give the reason as more like "rendering bug", rather than the id?

File: tests/ui/test_ui_tools.py [hunk 3]
```diff
@@ -1302,13 +1302,21 @@ def test_keypress_edit_message(self, mocker, message_fixture,
...
     ], ids=[
         "quoted level 1",
         "quoted level 2",
         "quoted level 3",
         "multi-line quoting",
-        "quoting with links"
```
Comment (neiljp) [inline tests/ui/test_ui_tools.py]: Trailing commas at the end of the last item means you don't get these diffs in future :)

File: tests/ui/test_ui_tools.py (hunk 1 above)
Comment (sumanthvrao) [reply tests/ui/test_ui_tools.py]: Agreed. That is a better description

File: tests/ui/test_ui_tools.py (hunk 2 above)
Comment (sumanthvrao) [reply tests/ui/test_ui_tools.py]: Done :+1:

File: tests/ui/test_ui_tools.py (hunk 3 above)
Comment (sumanthvrao) [reply tests/ui/test_ui_tools.py]: Thanks for sharing this :+1: (I had never though of this before)

**Comment by sumanthvrao** [comment]
@neiljp updated this with the correct failing tests. Will work on solving these as a follow-up.

**Comment by neiljp** [comment]
@sumanthvrao Thanks for following-through with this tests PR after the quotes indent feature :+1: this will be a good pointer to solving #402 and ensuring we don't have a regression on the status we currently have :)

**Pull request merged**